lib/dal/src/attribute/prototype/debug.rs

```diff
@@ -0,0 +1,207 @@
+
+                let func_arg_id =
+                    AttributePrototypeArgument::func_argument_id_by_id(ctx, apa_id).await?;
+                let func_arg_name = ctx
+                    .workspace_snapshot()?
+                    .get_node_weight_by_id(func_arg_id)
+                    .await?
+                    .get_func_argument_node_weight()?
+                    .name()
+                    .to_owned();
+
+                let values_for_arg =
+                    match AttributePrototypeArgument::value_source_by_id(ctx, apa_id)
+                        .await?
+                        .ok_or(
+                            AttributeValueError::AttributePrototypeArgumentMissingValueSource(
+                                apa_id,
+                            ),
+                        )? {
+                        ValueSource::StaticArgumentValue(static_argument_value_id) => {
+                            let val = StaticArgumentValue::get_by_id(ctx, static_argument_value_id)
+                                .await?
+                                .value;
+                            arg_sources.insert(
+                                func_arg_name.clone(),
+                                Some(PropPath::new(["Static Value"]).to_string()),
+                            );
+                            vec![val]
+                        }
+                        other_source => {
+                            let mut values = vec![];
+
+                            for av_id in other_source
+                                .attribute_values_for_component_id(
+                                    ctx,
+                                    expected_source_component_id,
+                                )
+                                .await?
+                            {
+                                let attribute_value = AttributeValue::get_by_id(ctx, av_id).await?;
+
+                                // no prop here, means it's from an input socket!
+                                let av_name = AttributeValue::is_for(ctx, av_id).await?;
+                                let prop_path: PropPath = match av_name {
+                                    ValueIsFor::InputSocket(id) => {
+                                        let inputsock = InputSocket::get_by_id(ctx, id).await?;
+                                        PropPath::new(["Input Socket", inputsock.name()])
+                                    }
+                                    ValueIsFor::Prop(_) => {
+                                        let prop_id = AttributeValue::prop_id_for_id(
+                                            ctx,
+                                            attribute_value.id(),
+                                        )
+                                        .await?;
+                                        Prop::path_by_id(ctx, prop_id).await?
+                                    }
+                                    _ => continue,
```

Comment:
nit: I think since there are only three `ValueIsFor` enum variants, I'd just explicitly write it here in case we add another one in the future and want the developer doing so to look at each location rather than getting masked by `_`. This is unlikely, but a good habit... depending on context.